File: src/main/java/io/github/dqualizer/dqlang/types/openapiadapter/APISchema.java
package io.github.dqualizer.dqlang.types.openapiadapter;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.github.dqualizer.dqlang.types.openapiadapter.data.DataSchema;
import io.github.dqualizer.dqlang.types.openapiadapter.field.FieldItem;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.ToString;

import java.util.Map;
import java.util.Set;

/**
 * Java class for the dqlang API-schema
 */
@Getter
@ToString
@AllArgsConstructor
@NoArgsConstructor
public class APISchema {

    private int version;
    private String context;
    private String api;
    @JsonProperty("server_info")
    private Set<ServerInfo> serverInfo;
    private Set<FieldItem> field;
    @JsonProperty("data_schemas")
    private Map<String, DataSchema> dataSchemas;
}

File: src/main/kotlin/io/github/dqualizer/dqlang/archive/openapiadapter/dqlang/APISchema.kt
package io.github.dqualizer.dqlang.archive.openapiadapter.dqlang

/**
 * Java class for the dqlang API-schema
 */
import com.fasterxml.jackson.annotation.JsonProperty
import io.github.dqualizer.dqlang.archive.openapiadapter.dqlang.data.DataSchema
import io.github.dqualizer.dqlang.archive.openapiadapter.dqlang.field.FieldItem
import lombok.AllArgsConstructor
import lombok.Getter
import lombok.NoArgsConstructor
import lombok.ToString

@Getter
@ToString
@AllArgsConstructor
@NoArgsConstructor
class APISchema {
    val version = 1
    val context: String? = null
    val api: String? = null

    @JsonProperty("server_info")
    val serverInfo: LinkedHashSet<io.github.dqualizer.dqlang.archive.openapiadapter.dqlang.ServerInfo>? = null
    val field: LinkedHashSet<io.github.dqualizer.dqlang.archive.openapiadapter.dqlang.field.FieldItem>? = null

    @JsonProperty("data_schemas")
    val dataSchemas: Map<String, io.github.dqualizer.dqlang.archive.openapiadapter.dqlang.data.DataSchema>? = null
}
